flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/table/catalog/client/PulsarCatalogClient.java

```diff
@@ -0,0 +1,406 @@
+            return false;
+        }
+
+        try {
+            String databaseName = configuration.databaseName(name);
+            admin.namespaces().getProperties(databaseName);
+
+            // No exception has been thrown means this is an existed database.
+            return true;
+        } catch (NotFoundException e) {
+            return false;
+        }
+    }
+
+    public void createDatabase(String name, CatalogDatabase database, boolean ignoreIfExists)
+            throws PulsarAdminException, DatabaseAlreadyExistException {
+        if (configuration.isInternalDatabase(name)) {
+            throw new CatalogException(name + " is a preserved Pulsar internal database");
+        }
+
+        String databaseName = configuration.databaseName(name);
+
+        try {
+            admin.namespaces().createNamespace(databaseName);
+        } catch (ConflictException e) {
+            // The database is already existed in Pulsar.
+            if (ignoreIfExists) {
+                // Skip the properties setting for an existed database.
+                return;
+            } else {
+                throw new DatabaseAlreadyExistException(catalogName, name, e);
+            }
+        }
+
+        Map<String, String> properties = databaseConverter.convert(database);
+        admin.namespaces().setProperties(databaseName, properties);
+    }
+
+    public void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade)
+            throws PulsarAdminException, DatabaseNotExistException, DatabaseNotEmptyException {
+        if (configuration.isInternalDatabase(name)) {
+            throw new CatalogException(
+                    name + " is an internal database, you don't have write access");
+        }
+
+        String databaseName = configuration.databaseName(name);
+
+        // List all the topics include the non-persistent topics for checking.
+        List<String> topics;
+        try {
+            ListNamespaceTopicsOptions options =
+                    ListNamespaceTopicsOptions.builder()
+                            .includeSystemTopic(false)
+                            .mode(Mode.ALL)
+                            .build();
+            topics = admin.namespaces().getTopics(databaseName, options);
+        } catch (NotFoundException e) {
+            if (ignoreIfNotExists) {
+                return;
+            } else {
+                throw new DatabaseNotExistException(catalogName, name, e);
+            }
+        }
+
+        // Check if this is an empty database.
+        if (!topics.isEmpty() && !cascade) {
+            throw new DatabaseNotEmptyException(catalogName, name);
+        }
+
+        // Delete all the tables under this database.
+        for (String topic : topics) {
+            LOG.warn("Delete table {} due to we want to remove the database {}", topic, name);
+            admin.topics().delete(topic, true);
```

Comment:
I am not particularly familiar with pulsar client. What will happen if it is not in the `force` mode.